### PR TITLE
lief: add version 0.14.1, update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **lief/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

**Comment by uilianries** [comment]
> C:\J2\w\prod-v2\bsr\3153\adfca\p\fmtc530f06ff8cce\p\include\fmt/ostream.h(32,1): fatal  error C1001: Internal compiler error. [C:\J2\w\prod-v2\bsr\3153\adfca\p\b\lief6d2927102a66d\b\build\LIB_LIEF.vcxproj]

I just restarted the Conan v2 Build ...

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by uilianries** [comment]
Commit conflict in the CI ... Restarted it again, sorry 😞

**Comment by toge** [comment]
Oh, It looks strange.

It was occurred in 0.13.1(still merged version).
```
C:\J2\w\prod-v2\bsr@2\4688\babfe\p\fmtc530f06ff8cce\p\include\fmt/ostream.h(32,1): fatal  error C1001: Internal compiler error. [C:\J2\w\prod-v2\bsr@2\4688\babfe\p\b\lief9d7c07d0fc10e\b\build\LIB_LIEF.vcxproj]
  (compiler file 'msc1.cpp', line 1589)
   To work around this problem, try simplifying or changing the program near the locations listed above.
```

**Comment by uilianries** [comment]
Just restarted it now. Hope it works.

File: recipes/lief/all/conanfile.py [hunk 1]
```diff
@@ -108,6 +108,11 @@ def validate(self):
         if self.settings.compiler.get_safe("libcxx") == "libstdc++":
             raise ConanInvalidConfiguration(f"{self.ref} does not support libstdc++")
 
+        # FIXME: Visual 2022 does has a bug: developercommunity.visualstudio.com/t/Internal-compiler-error-compiler-file-m/10376323
+        # fatal  error C1001: Internal compiler error. [C:\J2\w\prod-v2\bsr\14710\ccfaa\p\b\lief70856170ca89f\b\build\LIB_LIEF.vcxproj]
+        if is_msvc(self) and self.settings.compiler.version in ["17", "193"]:
```
Comment (uilianries) [inline recipes/lief/all/conanfile.py:113]:
```suggestion
        if is_msvc(self) and str(self.settings.compiler.version) in ["17", "193"]:
```

**Comment by gureedo** [comment]
PR #20568 should fix link error

**Comment by toge** [comment]
@RubenRBS 
Could you regenerate binary?

```
ERROR: Missing binary: spdlog/1.10.0:fc6983b74772bc0fec6c8c5701962c5554112e61
spdlog/1.10.0: WARN: Can't find a 'spdlog/1.10.0' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Debug, compiler=Visual Studio, compiler.runtime=MDd, compiler.version=16, os=Windows
- Options: header_only=False, no_exceptions=False, shared=False, wchar_filenames=False, wchar_support=False, fmt:header_only=False, fmt:shared=False, fmt:with_fmt_alias=False, fmt:with_os_api=True
- Dependencies: fmt/8.1.1
- Requirements: fmt/8.Y.Z
- Package ID: fc6983b74772bc0fec6c8c5701962c5554112e61
```

**Pull request closed without merging**